### PR TITLE
rename IsNonHTTPTLSServer to IsTLSServer

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/config/gateway/gateway.go [hunk 1]
```diff
@@ -20,8 +20,9 @@ import (
 	"istio.io/istio/pkg/config/protocol"
 )
 
-// IsNonHTTPTLSServer returns true if this server is non HTTP, but with some TLS settings for termination/passthrough
-func IsNonHTTPTLSServer(server *v1alpha3.Server) bool {
```
Comment (hzxuzhonghu) [inline pkg/config/gateway/gateway.go:24]: This is very unreadable

**Comment by ramaraochavali** [comment]
you only changed from IsTLSServer -> IsNonHttpTLSServer https://github.com/istio/istio/pull/38895 :laughing:

But if you prefer the old name - I will approve this.

**Comment by hzxuzhonghu** [comment]
This sounds confusing, lol

**Comment by ramaraochavali** [comment]
Are you OK with this changed name? I am fine with either one. Let me know. I can approve

**Comment by hzxuzhonghu** [comment]
Yep, this function especially exclude https redirect.

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**